### ISSUE TITLE
Could not find a declaration file for module 'vue-search-input'. Vue3 + Vite3

### ISSUE THREAD

**Issue by compilekaiten** [kouts/vue-search-input#10]
I'm using the [Vitesse](https://github.com/antfu/vitesse) template for my project. I wanted to incorporate this, but I keep getting this error.

As a new developer, I'm a bit confused on how I could make this work properly. 

Please let me know if you have any suggestions! Thanks

**Comment by kouts** [comment]
Hello @compilekaiten, I don't think using Vitesse makes any difference in the way you `import` and use a Vue component.
What's the error that you are getting, can you post a [Stackblitz](https://stackblitz.com/) example?

**Comment by compilekaiten** [comment]
The error I'm receiving is: 
```Could not find a declaration file for module 'vue-search-input'. 'C:/redacted/node_modules/.pnpm/vue-search-input@1.1.9/node_modules/vue-search-input/dist/vue-search-input.umd.js' implicitly has an 'any' type.```

Adding: `declare module 'vue-search-input'` to my `shims.d.ts` makes the error go away, but I don't get any typechecking/intellisense. 

Attempting to run `pnpm build`, I get the following error:
```
(node:13236) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.

SyntaxError: Cannot use import statement outside a module
```

Although my package.json type is set to module.
```
{
  "type": "module",
  "private": true,
  "packageManager": "pnpm@7.9.0",
  "scripts": {
    "build": "vite-ssg build",
    "dev": "vite --port 3333",
    "open": "vite --port 3333 --open",
    "lint": "eslint .",
    "preview": "vite preview",
    "preview-https": "serve dist",
    "test": "vitest",
    "test:e2e": "cypress open",
    "test:unit": "vitest",
    "typecheck": "vue-tsc --noEmit",
    "u": "taze major -I"
  },
  "dependencies": {
    "@bahmutov/cy-api": "^2.2.0",
    "@types/node": "*",
    "@vicgutt/tailwindcss-debug": "^0.1.0",
    "@vueuse/core": "^9.1.1",
    "@vueuse/head": "^0.7.9",
    "autoprefixer": "^10.4.8",
    "axios": "^0.27.2",
    "daisyui": "^2.24.0",
    "mobile-detect": "^1.4.5",
    "postcss": "^8.4.16",
    "swiper": "^8.3.2",
    "tailwind-scrollbar": "^1.3.1",
    "tailwindcss": "^3.1.8",
    "theme-change": "^2.2.0",
    "ts-node": "^10.9.1",
    "vue": "^3.2.38",
    "vue-demi": "^0.13.11",
    "vue-request": "2.0.0-rc.3",
    "vue-router": "^4.1.5",
    "vue-search-input": "^1.1.9"
  },
  "devDependencies": {
    "@antfu/eslint-config": "^0.26.3",
    "@iconify/json": "^2.1.102",
    "@vitejs/plugin-vue": "^3.0.3",
    "@vue/compiler-sfc": "^3.2.38",
    "@vue/test-utils": "^2.0.2",
    "critters": "^0.0.16",
    "cross-env": "^7.0.3",
    "cypress": "^10.7.0",
    "eslint": "^8.23.0",
    "eslint-plugin-cypress": "^2.12.1",
    "https-localhost": "^4.7.1",
    "pnpm": "^7.9.5",
    "taze": "^0.8.0",
    "typescript": "^4.8.2",
    "unplugin-auto-import": "^0.11.2",
    "unplugin-icons": "^0.14.9",
    "unplugin-vue-components": "^0.22.4",
    "vite": "3.0.9",
    "vite-plugin-inspect": "^0.6.1",
    "vite-plugin-pages": "^0.26.0",
    "vite-plugin-pwa": "^0.12.6",
    "vite-plugin-vue-layouts": "^0.7.0",
    "vite-ssg": "^0.20.2",
    "vite-ssg-sitemap": "^0.4.1",
    "vitest": "^0.22.1",
    "vue-tsc": "^0.40.6"
  }
}
```

My tsconfig.json is as follows:
```
{
  "compilerOptions": {
    "baseUrl": ".",
    "module": "ESNext",
    "target": "ESNext",
    "lib": ["DOM", "ESNext"],
    "strict": true,
    "esModuleInterop": true,
    "jsx": "preserve",
    "skipLibCheck": true,
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "noUnusedLocals": true,
    "strictNullChecks": true,
    // "allowJs": true,
    "forceConsistentCasingInFileNames": true,
    "types": [
      "vitest",
      "vite/client",
      "vue/ref-macros",
      "vite-plugin-pages/client",
      "vite-plugin-vue-layouts/client",
      "unplugin-icons/types/vue"
    ],
    "paths": {
      "~/*": ["src/*"]
    }
  },
  "exclude": ["dist", "node_modules", "cypress"]
}
```

**Comment by compilekaiten** [comment]
Here is an example that I made, copying most of my packages/etc.

Run `pnpm build` and it'll error out.

https://stackblitz.com/edit/vitejs-vite-bfomm1?file=src/App.vue

**Comment by kouts** [comment]
I've tested it and it indeed errors out. Maybe this is something Vitesse specific, I'll try with a vanilla setup when I find some time and let you know @compilekaiten

**Comment by kouts** [comment]
Fixed with #13

**Comment by compilekaiten** [comment]
Let's goooo! Thanks!
